protocol/v2/ssv/runner/committee.go

```diff
@@ -539,12 +539,14 @@ func findValidators(
 
 // Unneeded since no preconsensus phase
 func (cr CommitteeRunner) expectedPreConsensusRootsAndDomain() ([]ssz.HashRoot, phase0.DomainType, error) {
+	// TODO: spec uses a panic here since this function should never be used. Should we align?
```

Comment:
panic inside our code isn't a good practice. we could use logger.Fatal. anyway not worried about this as like you said, should never be used.

Reply:
Agree

Reply:
@MatheusFranco99 why isnt just returning an error enough? tests should fail if they reach here, unless they dont check error for some reason which is an error in the test already? 🤔

Reply:
Yeah, I agree